VPN UI improvements part 2 - new header animation

samsymons opened the pull request:
<!--
Note: This checklist is a reminder of our shared engineering expectations. Feel free to change it, although assigning a GitHub reviewer and the items in bold are required.

⚠️ If you're an external contributor, please file an issue first before working on a PR, as we can't guarantee that we will accept your changes if they haven't been discussed ahead of time. Thanks!
-->

Task/Issue URL: https://app.asana.com/0/414235014887631/1206747946345656/f
Tech Design URL:
CC:

**Description**:

This PR adds the new header animation to the iOS UI.

<!--
If at any point it isn't actively being worked on/ready for review/otherwise moving forward strongly consider closing it (or not opening it in the first place). If you decide not to close it, use Draft PR while work is still in progress or use `DO NOT MERGE` label to clarify the PRs state and comment with more information.
-->

**Steps to test this PR**:
1. Run the app and connect to NetP
2. Check that the header animation plays and looks rad
3. Leave NetP connected for a few seconds and check that an idle animation plays
4. Change the device theme and check that the asset changes
5. Stop NetP and check that the animation stops
6. Start NetP again and then back out of the status view controller
7. Go back into it and check that the animation is immediately playing the idle state, it doesn't show the connecting animation again

<!--
Before submitting a PR, please ensure you have tested the combinations you expect the reviewer to test, then delete configurations you *know* do not need explicit testing.

Using a simulator where a physical device is unavailable is acceptable.
-->

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

samsymons committed:
Add VPN header animation.

samsymons committed:
Stop the Lottie animation when the binding changes.

samsymons committed:
Allow the animation intro to be skipped

samsymons committed:
Add dark mode animation.

samsymons committed:
Merge branch 'sam/vpn-ui-improvements' into sam/vpn-ui-improvements-2-lottie-animation

* sam/vpn-ui-improvements:
  Point to BSK hotfix (#2558)
  Prevent redeeming invite codes after the VPN test ends (#2559)
  Updates Kingfisher to 7.11.0 (#2545)
  Makes network path metadata description anonymous (#2544)
  Adds pixels to track main VPN funnels (#2543)
  Cleanup after rolling out autoconsent enabled by default (#2537)
  Add #URL macro (#2540)
  Handle expired entitlement in NetP (#2525)
  Remove isSubscriptionEnabled check when attempting to delete NetP token (#2548)
  Validate correct environment (#2546)
  Remove CGNAT range (#2524)
  Add preliminary support for subscription keychain sharing (#2538)
  fix opening tabs with transitional (#2542)
  Integrate confirm entitlements endpoint (#2541)
  Makes dbSaveBloomFilterError daily and count (#2526)
  16. Subscription: Display "Activation in progress" message (#2535)
  Autofill support for deleting all passwords  (#2497)
  Bump BrowserServicesKit (#2532)
  Release 7.111.0-0 (#2534)
  if dax dialogs are showing then dismiss (#2506)

samsymons committed:
Merge branch 'sam/vpn-ui-improvements' into sam/vpn-ui-improvements-2-lottie-animation

* sam/vpn-ui-improvements:
  Subscriptions: 20 - Subscription Caching (#2569)
  Subscriptions: 19. Error handling and minor updates (#2567)
  Use History on iOS (#2539)
  Release 7.112.0-1 (#2573)
  Report on toggle protections off (#2536)
  Release 7.112.0-0 (#2572)
  Update autoconsent to v10.2.0 (#2554)
  update metadata (#2571)
  Subscription pro pixels (#2531)
  Update SwiftSoup and Kingfisher versions (#2566)
  Move vpnFirstEnabled and networkPathChange out of VPNSettings (#2560)
  Fix VPN view model memory leak (#2570)
  Bump submodules/privacy-reference-tests from `40ce868` to `a603ff9` (#2500)
  18. Subscription Entitlements caching (#2556)
  Release 7.111.0-2 (#2563)
  Revert "Report Apple Ad attribution using pixel (#2510)" (#2562)
  Vanilla browser integration (#2550)
  Fix blank space after URL bar hides (#2549)
  Release 7.111.0-1 (#2561)

samsymons commented on DuckDuckGo/LottieView.swift:
```diff
@@ -22,35 +22,71 @@ import Lottie
  
 struct LottieView: UIViewRepresentable {
     
+    struct LoopWithIntroTiming {
+        let skipIntro: Bool
+        let introStartFrame: AnimationFrameTime
+        let introEndFrame: AnimationFrameTime
+        let loopStartFrame: AnimationFrameTime
+        let loopEndFrame: AnimationFrameTime
+    }
```
Had to get creative with the LottieView so that we could play more involved animations like this one.

samsymons committed:
Merge branch 'sam/vpn-ui-improvements' into sam/vpn-ui-improvements-2-lottie-animation

* sam/vpn-ui-improvements:
  Fix unit tests.
  rollback #URL macro (#2585)
  Release 7.112.0-2 (#2579)
  Updates BSK (#2577)

samsymons committed:
Merge branch 'sam/vpn-ui-improvements' into sam/vpn-ui-improvements-2-lottie-animation

* sam/vpn-ui-improvements:
  Update wireguard-apple to 1.1.3 (#2598)
  Updates BSK to include latest changes (#2604)
  Fix VPN feedback form description (#2606)
  Fixes the name of a pixel (#2575)
  22. Subscription- Cleanup and minor updates (#2596)
  Release 7.112.0-3 (#2602)
  update embedded file
  Bump autoconsent to 10.3.0 (#2601)
  Roll back CPM post-rollout cleanup (#2599)
  Stub objects for Bookmarks DB (#2593)
  Subscriptions - 21. Manage Billing options to third parties (#2574)
  17. Subscription Review Fixes (#2547)
  bump bsk to fix retain cycle (#2595)
  New log for SKAd 4 integration (#2594)
  NetP x Subscription Clean-up (#2565)
  Update BSK to include latest macOS changes (#2588)

samsymons commented on DuckDuckGo/LottieView.swift:
```diff
@@ -22,35 +22,71 @@ import Lottie
     }
  
     func updateUIView(_ uiView: UIViewType, context: Context) {
+        if uiView.isAnimationPlaying, !isAnimating.wrappedValue {
+            uiView.stop()
+            return
+        }
```
If the animation is playing but the wrapped value is false, then stop the view from animating. Without this change, setting `isAnimating` to false did nothing.

quanganhdo reviewed (approved):
Love the new animation 👏

samsymons committed:
VPN UI improvements part 3 - settings updates (#2582)

Task/Issue URL: https://app.asana.com/0/0/1206747946345657/f
Tech Design URL:
CC:

Description:

This PR removes the notification settings UI and moves those changes into the main settings screen.
